File: digital-bank/src/main/java/com/bootcamp/digitalbank/services/UserService.java
package com.bootcamp.digitalbank.services;

import java.util.List;

import com.bootcamp.digitalbank.DAO.entities.*;
import com.bootcamp.digitalbank.DAO.repository.UserRepository;

import org.springframework.stereotype.Service;

@Service
public class UserService {

	private UserRepository userRepository;

	public UserService(UserRepository userRepository) {
		this.userRepository = userRepository;
	}

	public List<User> findAll() {
		return userRepository.findAll();
	}

	public User save(User user) {
		return userRepository.save(user);
	}

	public void deleteAll(){
		userRepository.deleteAll();
	}
    
}

File: nest+graphql+angular/server/src/users/user.service.ts
import { Injectable } from '@nestjs/common'
import { InjectRepository } from '@nestjs/typeorm'
import { User } from '../database/entities/user.entity'
import { UserDTO } from '../users/models/user.dto'
import { BaseService } from '../services/base.service'
import { DeleteResult, Repository } from 'typeorm'

@Injectable()
export class UserService implements BaseService<UserDTO> {
  constructor (@InjectRepository(User) private readonly userRepository: Repository<User>) { }

  public async save (entity: UserDTO): Promise<User> {
    return await this.userRepository.save(entity)
  }

  public async findAll (): Promise<UserDTO[]> {
    return await this.userRepository.find()
  }

  public async findOne (email: string): Promise<UserDTO> {
    return await this.userRepository.findOne({ where: { email } })
  }

  public async remove (id: number): Promise<boolean> {
    const result: DeleteResult = await this.userRepository.delete(id)
    return result.affected > 0
  }
}
